lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+const WebInspector = require('../../lib/web-inspector');
+
+/**
+ * @fileoverview Gets JavaScript file content
+ */
+class Scripts extends Gatherer {
+  /**
+   * @param {{driver: !Driver}} options
+   * @param {{networkRecords: !Array<WebInspector.NetworkRequest}} traceData
+   * @return {!Promise<!Map<string, string>>}
+   */
+  afterPass(options, traceData) {
+    const driver = options.driver;
+
+    const scriptContentMap = new Map();
+    const scriptRecords = traceData.networkRecords
+      .filter(record => record.resourceType() === WebInspector.resourceTypes.Script);
+
+    return scriptRecords.reduce((promise, record) => {
+      return promise
+        .then(() => {
+          return driver.getRequestContent(record.requestId)
+            .catch(_ => null)
+            .then(content => {
+              if (!content) return;
+              scriptContentMap.set(record.url, content);
```

Comment:
since we're definitely dealing with networkRecords i'd rather be using requestIds here as the key.  

over in the audit we could could then use `WebInspector.NetworkLog.requestForId()` to grab the request and pull the URL off that. wdyt?

i suppose this'll make testing slightly harder, so curious what you think.

Reply:
yeah that's fair, if same URL was requested multiple times with different content we should surface that 👍